### DISCUSSION TITLE
NUT client upssched.conf issue

### DISCUSSION THREAD

**Discussion by bogolese** [techno-tim/techno-tim.github.io#221] [Questions]
Hi Tim,

I used the upssched.conf from your video and my machine done shut itself down on me!  :)  The UPS went on battery briefly (1 second), which started two timers:  onbatt and earlyshutdown.  When the UPS came back on line ONLINE event canceled the onbatt timer, but the earlyshutdown timer continued until it timed out and shut my system down.  :(

Anyway, I have added another event:

AT ONLINE * CANCEL-TIMER earlyshutdown online

So that my entire file now looks like:

CMDSCRIPT /etc/nut/upssched-cmd
PIPEFN /etc/nut/upssched.pipe
LOCKFN /etc/nut/upssched.lock

AT ONBATT * START-TIMER onbatt 120
AT ONLINE * CANCEL-TIMER onbatt online
AT ONBATT * START-TIMER earlyshutdown 120
AT ONLINE * CANCEL-TIMER earlyshutdown online
AT LOWBATT * EXECUTE onbatt
AT COMMBAD * START-TIMER commbad 120
AT COMMOK * CANCEL-TIMER commbad commok
AT NOCOMM * EXECUTE commbad
AT SHUTDOWN * EXECUTE powerdown
AT SHUTDOWN * EXECUTE powerdown

p.s., I am really running standalone but set it up as client/server because I was going to have my VMs shut themselves down, but ended up just letting the host suspend them as part of the shutdown process -- hence the relatively short timer.

**Comment by bogolese** [comment]
BTW, this was the output from my Ubuntu 19 /var/log/syslog that put me onto this:

Sep 26 09:52:42 MSIC2022 upsmon[2878]: UPS apcsmartups1500@localhost on battery
Sep 26 09:52:42 MSIC2022 upssched[30200]: Timer daemon started
Sep 26 09:52:43 MSIC2022 upssched[30200]: New timer: onbatt (120 seconds)
Sep 26 09:52:43 MSIC2022 upssched[30200]: New timer: earlyshutdown (120 seconds)
Sep 26 09:52:43 MSIC2022 upsmon[2878]: UPS apcsmartups1500@localhost on line power
Sep 26 09:52:43 MSIC2022 upssched[30200]: Cancelling timer: onbatt
Sep 26 09:54:43 MSIC2022 upssched[30200]: Event: earlyshutdown 
Sep 26 09:54:43 MSIC2022 upssched-cmd: UPS on battery too long, early shutdown
Sep 26 09:54:43 MSIC2022 upsmon[2878]: Signal 10: User requested FSD
Sep 26 09:54:43 MSIC2022 upsmon[2878]: FSD set on UPS apcsmartups1500@localhost failed: ERR ACCESS-DENIED
Sep 26 09:54:43 MSIC2022 upsmon[2878]: Executing automatic power-fail shutdown
Sep 26 09:54:43 MSIC2022 upsmon[2878]: Auto logout and shutdown proceeding
Sep 26 09:54:43 MSIC2022 upssched[31124]: Executing command: powerdown
Sep 26 09:54:43 MSIC2022 upssched-cmd: Unrecognized command: powerdown
Sep 26 09:54:43 MSIC2022 upssched[31124]: Executing command: powerdown
Sep 26 09:54:43 MSIC2022 upssched-cmd: Unrecognized command: powerdown
Sep 26 09:54:48 MSIC2022 systemd[1]: nut-monitor.service: Succeeded.
Sep 26 09:55:48 MSIC2022 systemd[1]: unattended-upgrades.service: Succeeded.
Sep 26 09:56:18 MSIC2022 systemd[1]: Stopping Session c2 of user rbergmann.

**Comment by timothystewart6** [comment]
Thank you!  We'll probably want to update the docs at some point!
